test/CheckedC/generic-function/generic-func-call-info.c

```diff
@@ -0,0 +1,29 @@
+// Test to make sure that the generic function calls are correctly instantiated
+//
+// In CallExpr, the type of the DeclRefExpr should be completely instantiated,
+// by replacing all type variable types with specified type names in the list
+// of type names provided when calling generic function.
+//
+//
+// RUN: %clang_cc1 -ast-dump -fcheckedc-extension %s | FileCheck %s

```

Comment:
Could you add tests for array types used as parameters?   There are 2 cases to cover:
- Incomplete array types as parameters:
`_For_any(T) int f(T[] arr)`
- Complete array types as parameters:
`_For_any(T) int f(T[10] arr)`